src/Symfony/Bundle/SecurityBundle/DependencyInjection/Security/Factory/AnonymousFactory.php

```diff
@@ -44,16 +45,7 @@ public function create(ContainerBuilder $container, $id, $config, $userProvider,
 
     public function createAuthenticator(ContainerBuilder $container, string $firewallName, array $config, string $userProviderId): string
     {
-        if (null === $config['secret']) {
-            $config['secret'] = new Parameter('container.build_hash');
-        }
-
-        $authenticatorId = 'security.authenticator.anonymous.'.$firewallName;
-        $container
-            ->setDefinition($authenticatorId, new ChildDefinition('security.authenticator.anonymous'))
-            ->replaceArgument(0, $config['secret']);
-
-        return $authenticatorId;
+        throw new InvalidConfigurationException('The authenticator manager no longer has anonymous security.');
```

Comment:
Quotes around “anonymous” and maybe the call to action:

Please remove this option under the “main” firewall.

Probably once we get the lazy firewall thing working, this might need to say something like “remove the anonymous option under the main firewall and add lazy: true instead”.